doc/harmony-ap_docker_installation_guide.md

```diff
@@ -1,406 +1,1309 @@
+
+At this point, you have a functional Access Point container. Next, we'll delve into how to persist its data and configure it in more detail, so that your setup is robust and tailored to your needs.
+
+## 3 Data Persistence
+
+Running Access Point in Docker requires careful handling of persistent data. Access Point stores essential data on the filesystem, for example, configuration files, security keys, logs, and (depending on configuration) message payloads or JMS data. If you run a container without preserving these, you would lose important data each time the container is removed or recreated. This section explains how and what to persist, and the options for doing so.
+
+### 3.1 Why Persistence is Needed
+
+Docker containers are ephemeral by default; any changes to the container's filesystem are lost when the container is stopped. The Harmony Access Point, however, is a stateful application:
+
+- **Configuration files:** These include `domibus.properties` (the main configuration), XML files for plugins, keystore files for certificates (TLS and signing keys), truststore files, and Tomcat configuration (`server.xml`). These are generated or populated on first startup in the container's `/var/opt/harmony-ap/etc` directory.
+-	**Security material:** The private keys and certificates are stored in keystore files on disk (e.g., `tls-keystore.p12`, `ap-keystore.p12`, and their corresponding truststores). Losing these would invalidate your node's identity and ability to decrypt messages.
+-	**Message payloads and working files:** Harmony AP may store incoming message attachments (database only by default) or certain working data on disk (especially if not configured to use database storage for large objects). As message attachments could be written under the data directory if enabled, if those files disappear, you might lose the ability to resend or recover messages.
+-	**JMS data (ActiveMQ):** The embedded ActiveMQ broker uses a file-based storage (KahaDB) to persist messages and transactions. This by default resides on the container's filesystem under the `/var/opt/harmony-ap/work` directory. If not persisted, a container restart could wipe in-transit messages or subscription info. Persisting the JMS store is important for reliability, particularly in production.
+
+In summary, without using volumes, each restart would act like a fresh installation: new keys, empty config, etc., which is not acceptable in production. Therefore, **using Docker volumes or bind mounts for `/var/opt/harmony-ap` or configuring the container to use the database for persistence is essential for a stable deployment**. This ensures that all configuration, security material, and message state are retained across container restarts or upgrades.
+
+### 3.2 Using Volumes for Persistence (Standalone Mode)
+
+For a single-instance deployment, you can use either:
```

Comment:
In this section I would also rely more on the Docker documentation and include less detail in the Harmony documentation.